napari/layers/image/image.py

```diff
@@ -249,93 +253,96 @@ def __init__(
         # Trigger generation of view slice and thumbnail
         self._update_dims()
 
-    def _calc_data_range(self):
+    def _calc_data_range(self) -> Tuple[float, float]:
         if self.multiscale:
             input_data = self.data[-1]
         else:
             input_data = self.data
         return calc_data_range(input_data)
 
     @property
-    def dtype(self):
-        return self.data[0].dtype if self.multiscale else self.data.dtype
+    def dtype(self) -> np.dtype:
+        if self.multiscale:
+            return self.data[0].dtype
+        else:
+            return cast(ArrayLike, self.data).dtype
 
     @property
-    def data(self):
-        """array: Image data."""
+    def data(self) -> Union[ArrayLike, List[ArrayLike]]:
+        """Image data. Will be a list of arrays if self.multiscale"""
         return self._data
 
     @data.setter
-    def data(self, data):
+    def data(self, data: Union[ArrayLike, List[ArrayLike]]):
         self._data = data
         self._update_dims()
         self.events.data()
 
-    def _get_ndim(self):
+    def _get_ndim(self) -> int:
         """Determine number of dimensions of the layer."""
         return len(self.level_shapes[0])
 
-    def _get_extent(self):
+    def _get_extent(self) -> Tuple[Tuple[int, int], ...]:
```

Comment:
* Image `-> Tuple[Tuple[int, int], ...]`
* Labels - inherits `_get_extent()` from Image.
The return type of `_get_extent()` is still inconsistent between different layer types. Is this intentional?

* Points `-> List[Tuple[int, int], ...]`
* Shapes - ?
* Surface `-> Tuple[Tuple[int, int], ...]`
* Vectors `-> List[Tuple[int, int], ...]`

Reply:
not intentional, thanks for pointing it out!
So far with this PR, I've mostly tried _not_ to actually alter the types of anything... just to annotate them as they are.  But I also think it would be within scope to unify types like that (and this is a perfect example), agree?

(also, looks like I haven't done shapes yet!  thanks)

Reply:
> So far with this PR, I've mostly tried _not_ to actually alter the types of anything... just to annotate them as they are. But I also think it would be within scope to unify types like that (and this is a perfect example), agree?

It certainly seems in scope, yes. Although if you do decide to keep them separate, it'd be good to open a tracking issue for it.